Source/WTF/wtf/NativePromise.h

```diff
@@ -530,14 +530,10 @@ class NativePromise final : public NativePromiseBase, public ConvertibleToNative
 
             m_resolveValues[index] = std::forward<ResolveValueType_>(resolveValue);
             if (!--m_outstandingPromises) {
-                Vector<ResolveValueType> resolveValues;
-                resolveValues.reserveInitialCapacity(m_resolveValues.size());
-                for (auto&& resolveValue : m_resolveValues)
-                    resolveValues.append(WTFMove(resolveValue.value()));
-
-                m_producer->resolve(WTFMove(resolveValues));
+                m_producer->resolve(WTF::map(WTFMove(m_resolveValues), [](auto&& resolveValue) {
+                    return WTFMove(*resolveValue);
+                }));
                 m_producer = nullptr;
-                m_resolveValues.clear();
```

Comment:
Don't we still want to clear `m_resolveValues`? Otherwise, we should consider using `std::exchange(m_resolveValues, { })` instead of `WTFMove(m_resolveValues)` above.